app/upgrades/v20/upgrades.go

```diff
@@ -40,3 +58,24 @@ func SetDydxCommunityPoolTreasuryAddress(ctx sdk.Context, stakeIbcKeeper stakeib
 
 	return nil
 }
+
+// Migrates Outstanding Downtime for upgrade to ICS v4
+// https://github.com/cosmos/interchain-security/blob/release/v4.0.x/UPGRADING.md#v40x
+func MigrateICSOutstandingDowntime(ctx sdk.Context, ck ccvconsumerkeeper.Keeper) error {
+	downtimes := ck.GetAllOutstandingDowntimes(ctx)
+	for _, od := range downtimes {
+		consAddr, err := sdk.ConsAddressFromBech32(od.ValidatorConsensusAddress)
+		if err != nil {
+			return err
+		}
+		ck.DeleteOutstandingDowntime(ctx, consAddr)
+	}
+	return nil
+}
+
+// Migrates ICS Params to add the new RetryDelayParam
+func MigrateICSParams(ctx sdk.Context, ck ccvconsumerkeeper.Keeper) {
+	params := ck.GetConsumerParams(ctx)
```

Comment:
Nice - so this didn't work pre-SDK 47, and now it does?

Reply:
that's my guess. Although, I think it might be up to the specific module. For instance, Stride as a chain is on SDK 47 but I don't think stakeibc ever upgraded the param store to the new format (low confidence though)

Either way, we can figure out if this works right away through the dockernet IG tests